### PR TITLE
Refactor the //tool/release package

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

We've split different functionalities in the `//tool/release` package into three sub-packages:

1. `version` package: contains functionalities for manipulating the version file
2. `notes` package: contains functionalities for generating release notes
3. `deps` package: contains functionalities for validating dependencies

## What are the changes implemented in this PR?

Split `//tool/release` into three sub-packages `version`, `notes`, and `deps`

### CODE REVIEW

File: tool/release/notes/BUILD [hunk 1]
```diff
@@ -24,9 +24,9 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
 package(default_visibility = ["//visibility:public"])
```
Comment (lolski) [inline tool/release/notes/BUILD:24]: Rename the package from `createnotes` to `notes`.

File: tool/release/notes/BUILD [hunk 2]
```diff
@@ -24,9 +24,9 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
...
 
 kt_jvm_binary(
-    name = "bin",
+    name = "create",
```
Comment (lolski) [inline tool/release/notes/BUILD:27]: Improve the name of the target by renaming it to `create` .

File: tool/release/notes/BUILD [hunk 3]
```diff
@@ -24,9 +24,9 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
...
     srcs = glob(["*.kt"]),
-    main_class = "com.vaticle.dependencies.tool.release.createnotes.CreateNotesKt",
+    main_class = "com.vaticle.dependencies.tool.release.notes.NotesCreateKt",
```
Comment (lolski) [inline tool/release/notes/BUILD:29]: Update `main_class` field accordingly.

File: tool/release/notes/Commit.kt [hunk 4]
```diff
@@ -19,10 +19,10 @@
  * under the License.
  */
 
-package com.vaticle.dependencies.tool.release.createnotes
+package com.vaticle.dependencies.tool.release.notes
```
Comment (lolski) [inline tool/release/notes/Commit.kt:22]: Rename the package from `createnotes` to `notes`.

File: tool/release/notes/Common.kt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.vaticle.dependencies.tool.release.createnotes
+package com.vaticle.dependencies.tool.release.notes
```
Comment (lolski) [inline tool/release/notes/Common.kt:1]: Rename the package from `createnotes` to `notes`.

File: tool/release/notes/Note.kt [hunk 6]
```diff
@@ -19,16 +19,16 @@
  * under the License.
  */
 
-package com.vaticle.dependencies.tool.release.createnotes
+package com.vaticle.dependencies.tool.release.notes
```
Comment (lolski) [inline tool/release/notes/Note.kt:22]: Rename the package from `createnotes` to `notes`.

File: tool/release/notes/NotesCreate.kt [hunk 7]
```diff
@@ -19,7 +19,7 @@
  * under the License.
  */
 
-package com.vaticle.dependencies.tool.release.createnotes
+package com.vaticle.dependencies.tool.release.notes
```
Comment (lolski) [inline tool/release/notes/NotesCreate.kt:22]: Rename the package from `createnotes` to `notes`.

File: tool/release/notes/NotesCreate.kt [hunk 8]
```diff
@@ -28,7 +28,7 @@ import kotlin.text.Regex.Companion.escapeReplacement
 
 fun main(args: Array<String>) {
     val bazelWorkspaceDir = Paths.get(getEnv("BUILD_WORKSPACE_DIRECTORY"))
-    val githubToken = getEnv("CREATE_NOTES_TOKEN")
+    val githubToken = getEnv("NOTES_CREATE_TOKEN")
```
Comment (lolski) [inline tool/release/notes/NotesCreate.kt:31]: Rename the environment variable to be consistent with how where the tool is located, which is `//library/tool/notes:create`.

File: tool/release/notes/Version.kt [hunk 9]
```diff
@@ -19,7 +19,7 @@
  * under the License.
  */
 
-package com.vaticle.dependencies.tool.release.createnotes
+package com.vaticle.dependencies.tool.release.notes
```
Comment (lolski) [inline tool/release/notes/Version.kt:22]: Rename the package from `createnotes` to `notes`.

File: tool/release/version/Bump.kt [hunk 10]
```diff
@@ -1,4 +1,4 @@
-package com.vaticle.dependencies.tool.release
+package com.vaticle.dependencies.tool.release.version
```
Comment (lolski) [inline tool/release/version/Bump.kt:1]: Move the file into the new package `version`.

File: tool/release/version/BUILD [hunk 11]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (lolski) [inline tool/release/version/BUILD:1]: Create the BUILD file for the `version` package.

File: tool/release/version/BUILD [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
+
+kt_jvm_binary(
```
Comment (lolski) [inline tool/release/version/BUILD:20]: Create the target `version:bump`.

File: tool/release/BUILD [hunk 13]
```diff
@@ -18,23 +18,9 @@
...
-
-
-exports_files(["ValidateDeps.kt", "ValidateNodeJsDeps.kt", "ValidatePythonDeps.kt"])
 
 py_binary(
     name = "docs",
     srcs = ["docs.py"],
     main = "docs.py"
 )
-
```
Comment (lolski) [inline tool/release/BUILD:31]: Move the `bump` target into the new package `version`.

File: tool/release/BUILD [hunk 14]
```diff
@@ -18,23 +18,9 @@
 package(default_visibility = ["//visibility:public"])
 
 load("@vaticle_dependencies_ci_pip//:requirements.bzl", "requirement")
-load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary")
-
-
-exports_files(["ValidateDeps.kt", "ValidateNodeJsDeps.kt", "ValidatePythonDeps.kt"])
```
Comment (lolski) [inline tool/release/BUILD:24]: Move the validation scripts into the new package `deps`.

File: tool/release/deps/ValidateDeps.kt [hunk 15]
```diff
@@ -1,4 +1,4 @@
-package com.vaticle.dependencies.tool.release
+package com.vaticle.dependencies.tool.release.deps
```
Comment (lolski) [inline tool/release/deps/ValidateDeps.kt:1]: Move the file into the new package `deps`.

File: tool/release/deps/ValidateNodeJsDeps.kt [hunk 16]
```diff
@@ -1,4 +1,4 @@
-package com.vaticle.dependencies.tool.release
+package com.vaticle.dependencies.tool.release.deps
```
Comment (lolski) [inline tool/release/deps/ValidateNodeJsDeps.kt:1]: Move the file into the new package `deps`.

File: tool/release/deps/ValidatePythonDeps.kt [hunk 17]
```diff
@@ -1,10 +1,9 @@
-package com.vaticle.dependencies.tool.release
+package com.vaticle.dependencies.tool.release.deps
```
Comment (lolski) [inline tool/release/deps/ValidatePythonDeps.kt:1]: Move the file into the new package `deps`.

File: tool/release/deps/BUILD [hunk 18]
```diff
@@ -0,0 +1,18 @@
+#
```
Comment (lolski) [inline tool/release/deps/BUILD:1]: Create the BUILD file for the package `deps`.

**Review by flyingsilverfin — APPROVED** [review]
Looks good, `.notes.` looks a LOT nicer than `.createnotes.`!!

**Comment by lolski** [comment]
**Maintaining Good Architecture**

The PR was approved without any change request, but nevertheless, I would like to share some principles that I applied in creating this PR.

**The Habit**
The cohesiveness of the architecture of a given repository / package needs to be maintained continuously, and it's a good habit to refactor the package that we worked on. On this occasion, I had to rework the implementation of the `create-notes` script, and out of habit decided to give the whole `release` package a look to see if the architecture still makes sense.

**Methodology**

The `release` package had about 10 files and we found that most functionalities revolve around dependencies, release notes, and versioning. It is clear that these three concepts are the key concepts that we need to capture, and hence we do so by creating a subpackage for each of them. The resulting structure looks like this:

```
tool/
  release/
    deps/
      validate-java
      validate-python
      validate-nodejs
    notes/
      create
    version/
      bump
```

**Pull request merged**